src/utils/sortObjectKeys.ts

```diff
@@ -0,0 +1,33 @@
+import { Settings, SortOrder } from "../types/settings";
+
+const sortHandlers = {
+  [SortOrder.ASC]: (object: Record<string, any>) =>
+    Object.fromEntries(
+      Object.entries(object).sort(([a], [b]) => a.localeCompare(b))
+    ),
+
+  [SortOrder.DESC]: (object: Record<string, any>) =>
+    Object.fromEntries(
+      Object.entries(object).sort(([a], [b]) => b.localeCompare(a))
+    ),
+};
+
+function sortObjectKeys(obj: any, sort: SortOrder = SortOrder.NONE): any {
+  if (sort === SortOrder.NONE || typeof obj !== "object" || obj === null)
+    return obj;
+
+  if (Array.isArray(obj)) {
+    return obj.map((item) => sortObjectKeys(item, sort));
+  }
+
+  const sortedEntries = Object.entries(obj).map(([key, value]) => [
+    key,
+    sortObjectKeys(value, sort),
+  ]);
+
+  return sortHandlers[sort]
+    ? sortHandlers[sort](Object.fromEntries(sortedEntries))
+    : obj;
```

Comment:
This validation can be done at beginning, to avoid unneeded calculations.

Reply:
here you could do the verification to see if sort passed is valid and do early return if not is valid (or make a throw)
![2025-03-28-10:29:23-screenshot](https://github.com/user-attachments/assets/79e1c107-6ea5-412d-9ef8-062ca7a0b135)